src/v/archival/ntp_archiver_service.cc

```diff
@@ -1061,11 +1062,33 @@ ntp_archiver::schedule_single_upload(const upload_context& upload_ctx) {
     std::vector<ss::future<cloud_storage::upload_result>> all_uploads;
     all_uploads.emplace_back(upload_segment(upload, std::move(locks)));
     size_t tx_size = 0;
+    // Compacted segments never have tx-manifests because all batches
+    // generated by aborted transactions are removed during compaction.
     if (upload_ctx.upload_kind == segment_upload_kind::non_compacted) {
-        auto tx = co_await get_aborted_transactions(upload);
-        tx_size = tx.size();
-        if (!tx.empty()) {
-            all_uploads.emplace_back(upload_tx(upload, std::move(tx)));
+        auto handle_exception = [this, &all_uploads](ss::log_level level) {
+            _rtclog.log(
+              level,
```

Comment:
nit: if you want, here and below you can use `vlogl` to get file context

Reply:
I'll update this in next PR which is blocked by this